### PR TITLE
Xpress NLP solver implementation (#205)

### PR DESCRIPTION
**Pull request by joaquimg**

### CODE REVIEW

File: src/MOI/MOI_wrapper.jl [hunk 1]
```diff
@@ -2738,7 +2768,82 @@ function MOI.optimize!(model::Optimizer)
...
+                    MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(c, x), 0.0),
+                    c_set[1],
+                    );
+                end
+            end
+        
+
+            # Passing constraints to the solver
+            for i in 0:length(model.nlp_constraint_info)-1
+                Lib.XPRSnlpchgformulastring(model.inner, Cint(i), join(["+"," ",to_str(model.nlp_constraint_info[i+1].expression)]))
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:2834]: For realz? The API is to pass it as a string?

Comment (odow) [reply src/MOI/MOI_wrapper.jl:2834]: Ah. Seems like you can also pass tape:
https://www.fico.com/fico-xpress-optimization/docs/latest/solver/nonlinear/HTML/chapAPIFuncs.html?scroll=secAddNLPart

Comment (joaquimg) [reply src/MOI/MOI_wrapper.jl:2834]: yes, there is a tape. we were lazy though.

**Comment by odow** [comment]
Closing in favor of #255

**Pull request closed without merging**